storage-access.bs

```diff
@@ -310,17 +310,47 @@ The Storage Access API defines a [=policy-controlled feature=] identified by the
+
+Our privacy considerations are more challenging where default third-party cookies are already deprecated. The challenge is to decide when and how to permit the Storage Access API to be used to revert a cookie-less (or cookie-partitioned) iframe to a pre-deprecation state, giving it access to its [=unpartitioned data=].
+
+In an ideal case, an iframe would only be able to gain access to its [=unpartitioned data=] if:
+
+1. the user interacts with the iframe, giving it [=transient activation=]
+2. the iframe is permitted by the embedder to use the API
+3. the iframe is a [=secure context=]
+4. the user grants express, pairwise permission to the embeddee to use its cookies in the embedder
+5. the user is not inundated by requests for the "<code>storage-access</code>" permission so their express permission is not undermined by fatigue
+
+This specification requires the first three of implementers. This provides guarantees that the user is aware of the content of the iframe, the embedder has not opted out of the iframe's authentication, and the third-party cookies are not disclosed to network attackers, respectively.
+
+The last two points are in tension. In an ideal world, we would show a prompt to the user in every call to {{Document/requestStorageAccess()}}. But, this would allow pages to prompt the user so frequently as to put the last point at the discretion of the page– a state we find unacceptable.
 
 <figure id=example-prompt>
 <img src=images/storage-access-prompt.png
      alt="A modal dialog box which states 'Do you want to allow “video.example” to use cookies and website data while browsing “news.example”? This will allow “video.example” to track your activity.' and which has two buttons, “Don’t Allow” and “Allow”.">
 <figcaption>An example prompt which could be shown to the user when a site calls `document.`{{Document/requestStorageAccess()}}.</figcaption>
 </figure>
 
+Thus, the last two points represent a key point of compromise. We permit [=implementation-defined=] behavior on when to grant or deny requests for [=unpartitioned data=] without requiring user choice so long as they meet all other requirements. This compromise weakens the privacy guarantees of this proposal, specifically point 4, to a degree under the control of the implementer and gives the implementer the power to render it impossible to get storage access with this API. However, this has proven necessary to enable condition 5 to be possible given our implementers' differing stances on the compromise between these two points.
+
+An important avenue for improvement is the removal of the [=implementation-defined=] silent grants and denies. This would improve the developer experience using the API across platforms.
```

Comment:
I don't think we need to discuss future improvements to the spec here, but not a strong opinion. Again, I think we should rather point to those compromises and advise user agents on how to handle them best.

Reply:
I'm willing to drop it, but do think to brings up a developer experience loss that was made with this compromise.

Reply:
Phrasing this as an encouragement on user agents to do better on this makes more sense to me as well.